packages/task/src/task.ts

```diff
@@ -37,11 +37,11 @@ export const initialState = Symbol();
 
 export type TaskStatus = (typeof TaskStatus)[keyof typeof TaskStatus];
 
-export type StatusRenderer<R> = {
-  initial?: () => unknown;
-  pending?: () => unknown;
-  complete?: (value: R) => unknown;
-  error?: (error: unknown) => unknown;
+export type StatusRenderer<R, D extends ReadonlyArray<unknown>> = {
```

Comment:
For backwards compatibility: in case someone is using this type directly in an annotation, I think we need a default value for `D` so it can be omitted:

```suggestion
export type StatusRenderer<R, D extends ReadonlyArray<unknown> = ReadonlyArray<unknown>> = {
```